cmd/object-api-errors.go

```diff
@@ -31,9 +31,16 @@ func toObjectErr(err error, params ...string) error {
 	if err == nil {
 		return nil
 	}
-	if errors.Is(err, context.Canceled) {
+
+	// Unwarp the error first
+	if werr := errors.Unwrap(err); werr != nil {
+		err = werr
+	}
+
+	if err == context.Canceled {
```

Comment:
`errors.Is` is recursive. This is not - not sure if I understand the reason for the change - seems to make it worse.

Reply:
Ah, ok I see what you are doing. Since we will be using it several places, let's add a helper:

```Go
// unwrapAll will unwrap the returned error completely.
func unwrapAll(err error) error {
	for {
		werr := errors.Unwrap(err)
		if werr == nil {
			return err
		}
		err = werr
	}
}
```